neqo-transport/src/connection.rs

```diff
@@ -3694,4 +3737,70 @@ mod tests {
         let c_tx_dgrams = send_bytes(&mut client, 0, now);
         assert_eq!(c_tx_dgrams.len(), 4);
     }
+
+    #[test]
+    fn test_discarded_initial_keys() {
+        qdebug!("---- client: generate CH");
+        let mut client = default_client();
+        let init_pkt_c = client.process(None, now());
+        assert!(init_pkt_c.as_dgram_ref().is_some());
+        assert_eq!(init_pkt_c.as_dgram_ref().unwrap().len(), 1232);
+
+        qdebug!("---- server: CH -> SH, EE, CERT, CV, FIN");
+        let mut server = default_server();
+        let init_pkt_s = server.process(Some(init_pkt_c.as_dgram_ref().unwrap().clone()), now());
+        assert!(init_pkt_s.as_dgram_ref().is_some());
+        assert_eq!(1, server.stats().packets_rx);
+        assert_eq!(0, server.stats().dups_rx);
+
+        qdebug!("---- client: cert verification");
+        let out = client.process(Some(init_pkt_s.as_dgram_ref().unwrap().clone()), now());
+        assert!(out.as_dgram_ref().is_some());
+
+        assert_eq!(2, client.stats().packets_rx);
+        assert_eq!(0, client.stats().dups_rx);
```

Comment:
Should we be recording drops as well as dups?  That might make this easier to test.